vars/cloudFoundryDeploy.groovy

```diff
@@ -240,8 +258,14 @@ private void handleMTADeployment(Map config, script) {
         .addIfEmpty('mtaPath', config.mtaPath ?: findMtar())
         .use()
 
-    dockerExecute(script: script, dockerImage: config.dockerImage, dockerWorkspace: config.dockerWorkspace, stashContent: config.stashContent) {
-        deployMta(config)
+    withCredentials([usernamePassword(
```

Comment:
Why have you moved this withCredentials out of the deploy method? 
You can also have multiple nested withCredentials